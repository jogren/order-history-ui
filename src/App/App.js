import React, { Component } from 'react';
import Container from '../Container/Container';
import Form from '../Form/Form';
import { getOrders } from '../apiCalls/apiCalls';
import './App.css';

class App extends Component {
  constructor() {
    super();
    this.state = {
      orders: [],
      error: ''
    }
  }

  componentDidMount() {
    getOrders()
      .then(data => this.setState({ orders: data }))
      .catch(error => this.setState({ error: 'There was a problem fetching your data.' }))
  }

  addPurchase = (e, formState) => {
    e.preventDefault();
    this.setState({ orders: [...this.state.orders, formState] })
  }

  render() {
    const { orders, error } = this.state;
    console.log(orders)
    return (
      <main className="App">
        <header>
          <h1 className='app-title'>My Order History Test</h1>
          <Form addPurchase={this.addPurchase}/>
        </header>
        <Container orders={orders}/>
      </main>
    );
  }
}

export default App;
