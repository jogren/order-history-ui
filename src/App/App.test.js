import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import { shallow } from 'enzyme';

describe('App', () => {
  let wrapper;
  let mockOrder;

  beforeEach(() => {
    wrapper = shallow(<App />)
    mockOrder = {
      id: 1,
      name: 'iPhone',
      description: 'It\'s the new iPhone 18',
      price: 800,
      img: 'https://store.storeimages.cdn-apple.com/4982/as-images.apple.com/is/iphone-xs-max-gold-select-2018?wid=940&hei=1112&fmt=png-alpha&qlt=80&.v=1550795409154'
    }
  })

  it('should keep an up-to-date snapshot of App', () => {
    expect(wrapper).toMatchSnapshot();
  })

  it.skip('should add an order when addPurchase is fired', () => {
    let expected = [mockOrder];
    let e = {
      preventDefault: jest.fn()
    }
    expect(wrapper.state('orders')).toEqual([]);
    wrapper.instance().addPurchase(e, mockOrder);
    expect(wrapper.state('orders')).toEqual(expected);
  })

  it('should delete an order when deletePurchase is fired', () => {
    wrapper.setState({ orders: [mockOrder] })
    let expected = [];
    // let deleteOrder(1) = jest.fn();
    expect(wrapper.state('orders')).toEqual([mockOrder]);
    wrapper.instance().deletePurchase(1);
    expect(wrapper.state('orders')).toEqual([]);
  })
})
