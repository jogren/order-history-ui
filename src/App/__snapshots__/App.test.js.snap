// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should keep an up-to-date snapshot of App 1`] = `
<main
  className="App"
>
  <header>
    <h1
      className="app-title"
    >
      My Order History Test
    </h1>
    <Form
      addPurchase={[Function]}
    />
  </header>
  <Container
    deletePurchase={[Function]}
    orders={Array []}
  />
</main>
`;
